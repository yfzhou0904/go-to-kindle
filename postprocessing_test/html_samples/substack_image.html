<!DOCTYPE html>
<html>
<head><title>Substack Image Article</title></head>
<body>
<p>Intro text before image.</p>
<div>
  <figure><a target="_blank" href="https://substackcdn.com/image/fetch/placeholder.png">
      <div>
        <picture>
          <source type="image/webp" srcset="https://substackcdn.com/image/fetch/placeholder.png 424w, https://substackcdn.com/image/fetch/placeholder.png 848w">
          <img src="https://substackcdn.com/image/fetch/placeholder.png" width="1456" height="762"
            data-attrs='{"src":"https://substack-post-media.s3.amazonaws.com/public/images/sample.png","srcNoWatermark":null}'
            alt="Substack image" srcset="https://substackcdn.com/image/fetch/placeholder.png 424w, https://substackcdn.com/image/fetch/placeholder.png 848w" sizes="100vw" loading="lazy" />
        </picture>
      </div>
    </a></figure>
</div>
<p>More text after image.</p>
</body>
</html>
