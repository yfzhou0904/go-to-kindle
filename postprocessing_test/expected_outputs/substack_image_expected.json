{
  "testName": "substack_image",
  "inputFile": "html_samples/substack_image.html",
  "baseURL": "https://example.com/substack",
  "excludeImages": false,
  "expectedImageCount": 1,
  "mustNotContain": [
    "srcset",
    "loading",
    "data-attrs"
  ],
  "mustContain": [
    "Intro text before image.",
    "data:image/png;base64,"
  ]
}
